### DISCUSSION TITLE
Shader code syntax in script editor for VisualShaderNodeCustom

### DISCUSSION THREAD

**Discussion by Foyezes** [godotengine/godot-proposals#11397] [Editor]
I've been making some custom visual shader nodes. Because the shader code is put in as string, I need to detect the errors manually. (I'm new at both shader code and gdscript). It would be nice to have error detection for VisualShaderNodeCustom.

![image](https://github.com/user-attachments/assets/c21e61c6-d9d9-4caf-b8a1-aaf4edcf85b5)

**Comment by Calinou** [comment]
Unfortunately, I'm not sure how this could be made to work. The script editor is not aware that the multiline string is meant to be shader code.

Other editors run into similar issues, e.g. when there's embedded HTML in a string. Sometimes, they manage to figure it out (perhaps by reading the string and putting it through various parsers with a resulting "score" for each language). This is bound to take quite a lot of resources in large files though.

**Reply by Mickeon** [reply]
We could have a special literal so symbolize a String to be highlighted in shader language but... that's definitely unusual, it'd be creating a direct dependency between GDScript and Godot's shading language.
